### ISSUE TITLE
Vertical suggestions list

### ISSUE THREAD

**Issue by sspanak** [sspanak/tt9#28]
1. The current suggestions list is somewhat counterintuitve, because the it is horizontal, but suggestions are selected with Up and Down buttons.
2. Also, it covers parts of the main application. This is extremely inconvenient in chat clients, where the input field is at the bottom (Messenger, Telegram, WhatsApp and others), because the users cannot see what they have actually typed.

To solve the issues above:
- Make the suggestions list vertical.
- Rearrange the bottom bar and put it there
  - Remove the OK button.
  - The Settings and Backspace buttons should remain on the right and should become as small as possible.
  - Put the suggestion list on the left and make it as large as possible.
  - Increase the height of the bottom bar, to fit at least 4 suggestions
  - ~~Ensure the suggestion list matches the look and feel of the bottom bar.~~ - Done in #90 

After #15 and #19.

**Comment by postpersonality** [comment]
Hi! I have few suggestions. Not sure if it is better to post them here or another thread.

- It would be nice to have different color schemes for suggestions (I'd prefer black-grey-white). 
- On my Qin F21 there is dedicated menu button (left soft button) which can be used to access TT9 settings
- Thus there is no need to show the on-screen buttons and it would be nice to have them optional to save some screen space

**Comment by sspanak** [comment]
Hi!

- I am definitely changing the default color scheme to black and white. I know the the entire look and feel of the application needs to be unified, but I haven't had the time to do it. However, customization will require a new settings screen, because the current one is not flexible enough to accomodate such options.
- The Settings screen is accessible by holding the star key, so it is possible to remove the on-screen button even now.
- I am still hesitating what to do with these buttons. I do want to save screen space, but Backspace is mandatory, because the hardware key does not work in Spotify and Firefox. And just for aesthetics and symmetry, I am keeping them both.

**Comment by TomJansen** [comment]
About the vertical suggestions list, I would prefer it to be an option to switch between horizontal and vertical.

**Comment by postpersonality** [comment]
Ok, then we can use menu button (left soft) as additional backspace and leave back button as back action. Of course that should be optional.

**Comment by postpersonality** [comment]
I really want to join development. I'm not familiar with Android but I can do Java stuff/logic stuff. Can you please give me one non-critical/non-urgent issue to start?

**Comment by sspanak** [comment]
I give up on making it vertical. I can't come up with a proper layout for that. It takes too much space and it is very inconvenient when viewing the punctuation characters, because one could only see a few of them. I find it much more usable to see the entire character list and being able to easily select both from the start and the end of the it.

Here is a screenshot of the work in progress:
![vertical](https://user-images.githubusercontent.com/27846105/198971365-d759ad38-e55e-4b02-a5d6-166489c4dcfa.png)

So, for now, let's stick with a horizontal list and just fulfill #90.

**Comment by sspanak** [comment]
> About the vertical suggestions list, I would prefer it to be an option to switch between horizontal and vertical.

No need to worry about it anymore.

> Ok, then we can use menu button (left soft) as additional backspace and leave back button as back action. Of course that should be optional.

Sure we can. Once I get to #35, in the Settings screen, I'll add dropdown menus for configuring the keypad shortcuts. I takes time, though...

> I really want to join development. I'm not familiar with Android but I can do Java stuff/logic stuff. Can you please give me one non-critical/non-urgent issue to start with?

I have no experience either and I have no idea what I am doing. :smile: I have just started learning. 

Unfortunately, the thing I would like to get a hand with is remaking the Settings screen, but that's heavily dependent on the SharedPreferences API.

**Comment by sspanak** [comment]
@postpersonality, see my comments in #83.

**Comment by postpersonality** [comment]
> So, for now, let's stick with a horizontal list 

What you think about making it multiline?
I've made quick scketch in GIMP:
![2-line](https://user-images.githubusercontent.com/2012215/198981384-de8054cc-c409-47e7-9041-17e6f0461001.png)

However this could puzzle the user on how to navigater over it with HW keyboard.

**Comment by sspanak** [comment]
> However this could puzzle the user on how to navigater over it with HW keyboard.
Right. This is how the built-in T9 keyboard works and I find it the least intuitive solution of all...

Currently, I would like to focus on making TT9 better usable, so I'll just attach the suggestions to the bottom bar, to prevent them cover the application on top. Usability is more important than pretty design to me, so I don't intend to work so hard on the latter.

**Comment by TomJansen** [comment]
>From what I know, it is not possible to attach the suggestions to the input field
I think I have found a way to do this, see https://stackoverflow.com/questions/4930416/android-edittexts-cursor-coordinates-absolute-position

To make this work, we need to get the x,y coordinates of the cursor in the input field and then float the suggestions under or above the cursor (depending on the position) in a vertical list. This functionality would almost be identical to the Nokia 3310
